packages/angular_devkit/build_angular/src/builders/extract-i18n/application-extraction.ts

```diff
@@ -0,0 +1,164 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import type { ɵParsedMessage as LocalizeMessage } from '@angular/localize';
+import type { MessageExtractor } from '@angular/localize/tools';
+import type { BuilderContext, BuilderOutput } from '@angular-devkit/architect';
+import assert from 'node:assert';
+import nodePath from 'node:path';
+import { buildApplicationInternal } from '../application';
+import type { ApplicationBuilderInternalOptions } from '../application/options';
+import { buildEsbuildBrowser } from '../browser-esbuild';
+import type { NormalizedExtractI18nOptions } from './options';
+
+export async function extractMessages(
+  options: NormalizedExtractI18nOptions,
+  builderName: string,
+  context: BuilderContext,
+  extractorConstructor: typeof MessageExtractor,
+): Promise<{
+  builderResult: BuilderOutput;
+  basePath: string;
+  messages: LocalizeMessage[];
+  useLegacyIds: boolean;
+}> {
+  const messages: LocalizeMessage[] = [];
+
+  // Setup the build options for the application based on the browserTarget option
+  const buildOptions = (await context.validateOptions(
+    await context.getTargetOptions(options.browserTarget),
+    builderName,
+  )) as unknown as ApplicationBuilderInternalOptions;
+  buildOptions.optimization = false;
+  buildOptions.sourceMap = { scripts: true, vendor: true };
+  if (buildOptions.ssr) {
+    buildOptions.ssr = false;
+  }
```

Comment:
You would also need to disable prerendering and app-shell as disabling ssr will only disable 
server side rendering but not the server build.

That said, in future we need to improve this, as there might be server only components that require translation such as the app-shell. See: https://github.com/angular/angular-cli/issues/22871

```suggestion
 buildOptions.ssr = false;
 buildOptions.appShell = false;
 buildOptions.prerender = false;
```